GraphClientFactory.md

```diff
@@ -18,11 +18,31 @@ There are two primary usage scenarios for the client factory.  Developers who wi
   - Set the SdkVersion header with the appropriate moniker based on the following structure `graph-{lang}-{version}`.  
 - Enable the developer to select a supported sovereign cloud using an enumerated list.  Selecting the sovereign cloud should ensure that the AuthenticationProvider uses the appropriate Authentication Endpoint.
 - Enable a developer to configure a HTTP proxy that will be used for outgoing requests.
-
+- Enable a developer to provide custom hosts. 
+  - `CustomHosts` option should be set on client creation.
+  - `CustomHosts` option should be made available in the `Context` so that it is available to the middleware during request processing.
+  - These URLs are different from the Graph service endpoints on the national clouds. 
+  - Certain workloads error out when an unexpected header is present in the request. The middlewares should check: 
+    - If the request URL is a Graph service endpoint or a custom host provided by the developer, then append request headers or modify the request content.
+    - Else the middleware should delete request headers added by that middleware.
+  - Example of an workload error - [LargeFileUploadTask upload to OneDrive caused CORS error due `SDKVersion` telemetry header](https://github.com/microsoftgraph/msgraph-sdk-javascript/issues/265)
+  - Provide capabilities to modify or update the `CustomURLs` option after the client creation.  
+ 
 ## Performance Considerations
 
 - If available on the platform, enable gzip compression of requests and responses.
 
+## Future Implementation
+
+- Enable developers to set `CustomHosts`. The progress can be tracked as follows:
+```
```

Comment:
nit: no need for triple ticks when using tables

Reply:
The triple ticks are to keep the table aligned.